### PR TITLE
Hardhat zksync chai matchers

### PR DESCRIPTION
**Pull request by mpopovac-txfusion**

### CODE REVIEW

**Comment by ly0va** [comment]
@popzxc I know you've been doing similar stuff for our new internal test framework, please review and give your opinion. Also, see if we can use some of this for our needs as well.

**Comment by popzxc** [comment]
Do I get it right that this is a port of (a subset of) [`@ethereum-waffle/chai`](https://www.npmjs.com/package/@ethereum-waffle/chai)?
If that's correct, @mpopovac-txfusion could you please roughly describe the differences between two?

**Comment by mpopovac-txfusion** [comment]
> Do I get it right that this is a port of (a subset of) [`@ethereum-waffle/chai`](https://www.npmjs.com/package/@ethereum-waffle/chai)? If that's correct, @mpopovac-txfusion could you please roughly describe the differences between two?

Hi @popzxc. This is actually extending new hardhat plugin called [hardhat-chai-matchers](https://hardhat.org/hardhat-chai-matchers/docs/overview) since [@ethereum-waffle/chai](https://www.npmjs.com/package/@ethereum-waffle/chai) is being [depricated](https://hardhat.org/hardhat-chai-matchers/docs/migrate-from-waffle) (hardhat-chai-matchers adds support for custom errors...).  Here I'm overriding a subset of the matchers in order to keep the same behavior. 
Differences:
- Responses from transactions that reverts within the zkSync doesn't have the same structure, thus impacting all matchers starting with reverted. 
- TxFee calculated from txReceipt (gasPrice and gasUsed) is not valid, because gasUsed doesn't match with the one obtained from zkSync. 
- Support for wallet transfer method with overrides.

File: packages/hardhat-zksync-chai-matchers/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# hardhat-zksync-chai-matchers
+
+[Hardhat](https://hardhat.org/) plugin that adds zkSync-specific capabilities to the [Chai](https://chaijs.com/) assertion library, making your smart contract tests easy to write and read.
```
Comment (popzxc) [inline packages/hardhat-zksync-chai-matchers/README.md:3]: I would like to see the documentation for the provided matchers.
Either here or in our public docs (then the link to the public documentation should be present here).

File: packages/hardhat-zksync-chai-matchers/src/constants.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export const PLUGIN_NAME = '@matterlabs/hardhat-zksync-chai-matchers';
```
Comment (popzxc) [inline packages/hardhat-zksync-chai-matchers/src/constants.ts:1]: Nitpick, but normally we add newlines at the end of the file. Please do so to be consistent with the company code style.

**Review by popzxc — APPROVED** [review]
Nice.

File: packages/hardhat-zksync-chai-matchers/src/constants.ts (hunk 2 above)
Comment (githubdoramon) [reply packages/hardhat-zksync-chai-matchers/src/constants.ts:1]: @popzxc do we have lint config files that could (should) be used in all projects?

File: packages/hardhat-zksync-chai-matchers/README.md (hunk 1 above)
Comment (githubdoramon) [reply packages/hardhat-zksync-chai-matchers/README.md:3]: We definitely need to work on the documentation for all the updated plugins :)

File: packages/hardhat-zksync-chai-matchers/src/constants.ts (hunk 2 above)
Comment (mpopovac-txfusion) [reply packages/hardhat-zksync-chai-matchers/src/constants.ts:1]: > @popzxc do we have lint config files that could (should) be used in all projects?

Found it.

**Pull request merged**